src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -695,29 +695,40 @@ export class DataViewsService {
   }
 
   /**
-   * Returns the default data view as an object. If no default is found, or it is missing
+   * Returns the default data view as an object.
+   * If no default is found, or it is missing
    * another data view is selected as default and returned.
+   * If no possible data view found to become a default returns null
+   *
    * @returns default data view
    */
+  async getDefaultDataView(): Promise<DataView | null> {
+    const patterns = await this.getIdsWithTitle();
+    let defaultId: string | undefined = await this.config.get('defaultIndex');
+    const exists = defaultId ? patterns.some((pattern) => pattern.id === defaultId) : false;
 
-  async getDefaultDataView() {
-    const patterns = await this.getIds();
-    let defaultId = await this.config.get('defaultIndex');
-    let defined = !!defaultId;
-    const exists = patterns.includes(defaultId);
-
-    if (defined && !exists) {
+    if (defaultId && !exists) {
       await this.config.remove('defaultIndex');
-      defaultId = defined = false;
+      defaultId = undefined;
     }
 
-    if (patterns.length >= 1 && (await this.hasUserDataView().catch(() => true))) {
-      defaultId = patterns[0];
+    if (!defaultId && patterns.length >= 1 && (await this.hasUserDataView().catch(() => true))) {
```

Comment:
The main bug here was missing `!defaultId` 

We don't want to override `defaultId` if we already have one and such data view exists